File: server/ITI-graduation-project-Advilaw-main/AdviLaw.Application/Features/AppointmentSection/DTOs/AppointmentDetailsDTO.cs
using AdviLaw.Domain.Entites.ScheduleSection;

namespace AdviLaw.Application.Features.AppointmentSection.DTOs
{
    public class AppointmentDetailsDTO
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }  
        public ScheduleType Type { get; set; }
        public ScheduleStatus Status { get; set; }
        public int JobId { get; set; }
        public int? ScheduleId { get; set; }

        // Extra for UI:
        public string JobHeader { get; set; } = string.Empty;
        public string ClientName { get; set; } = string.Empty;
        public string LawyerName { get; set; } = string.Empty;
    }
}

File: client/ITI-graduation-project-Advilaw-Frontend-main/src/app/types/Appointments/AppointmentDetailsDTO.ts
import { AppointmentStatus } from './AppointmentStatus';
import { AppointmentType } from './AppointmentType';

export interface AppointmentDetailsDTO {
  id: number;
  date: string;
  type: AppointmentType;
  status: AppointmentStatus;
  jobId: number;
  scheduleId?: number;
  
  // Extra for UI:
  jobHeader: string;
  clientName: string;
  lawyerName: string;
}
